truffleruby/src/main/java/org/truffleruby/platform/posix/JNRTrufflePosix.java

```diff
@@ -446,28 +410,77 @@ public int open(CharSequence path, int flags, int perm) {
     @TruffleBoundary
     @Override
     public int write(int fd, byte[] buf, int n) {
+        if ((fd == 1 || fd == 2) && context.getOptions().POLYGLOT_STDIO) {
```

Comment:
Options-based conditions are normally written the other way around to clarify the condition is not evaluated if the option is not enabled:
```java
 if (context.getOptions().POLYGLOT_STDIO && (fd == 1 || fd == 2)) {
```
Could you change that?